backend/src/middleware/auth.js

```diff
@@ -0,0 +1,46 @@
+const keyPath =
+  process.env.NODE_ENV === 'test' ? '../../tests/artifacts/test-keys.json' : process.env.KEY_JSON
+const keyData = require(keyPath)
+
+const auth = (req, res, next) => {
+  const key = req.query.key
+  if (key != undefined) {
+    if (key && key.length === 11) {
+      const keysFiltered = keyData.keys.filter(currKey => currKey.key === key)
+      keyVerified = keysFiltered.length !== 0
+
+      if (keyVerified) {
+        req._key_name = keysFiltered[0].name // set the user's name of the key that was used to make the request
+        req._key = key
+
+        // check if there is a corresponding name for the key
+        if (req._key_name == undefined) {
+          msg = `The key '${key}'given doesn't have a corresponding name. Check the json file holding the keys.`
+          console.error(msg)
+          res.status(500).json({
+            code: 500,
+            message: 'Bad Key',
+            success: false
+          })

```

Comment:
Right now, I think this would hit the `return next()` line, which we don't want if we can't get the name. Can we `return` after setting the status to 500?

Reply:
just fixed it